pkg/server/endpoints/endpoints.go

```diff
@@ -338,7 +354,20 @@ func (e *Endpoints) runLocalAccess(ctx context.Context, server *grpc.Server) err
 		return err
 	case <-ctx.Done():
 		log.Info("Stopping Server APIs")
-		server.Stop()
+		stopComplete := make(chan struct{})
+		go func() {
+			log.Info("Attempting graceful stop")
+			server.GracefulStop()
+			close(stopComplete)
+		}()
+
+		shutdownDeadline := time.After(e.GracefulStopWait)
+		select {
+		case <-shutdownDeadline:
+			log.Infof("Graceful stop unsuccessful, forced stop after %s", e.GracefulStopWait.String())
+		case <-stopComplete:
+			log.Info("Graceful stop successful")
+		}
```

Comment:
Can we have a helper function for this, since it's done in 2 different places?

Reply:
added... I had considered this originally but didn't like the idea of having to pass the parameterized logger around. what do you think of the implementation? an alternative would be to reconstruct a new logger in the helper but that would require passing the listener/information about the listener into the helper